integrationtests/self/http_test.go

```diff
@@ -140,6 +140,24 @@ var _ = Describe("HTTP tests", func() {
 		Expect(resp.Header.Get("Content-Length")).To(Equal(strconv.Itoa(len("foobar"))))
 	})
 
+	It("detects stream errors when server panics when writing response", func() {
+		mux.HandleFunc("/writing_and_panicking", func(w http.ResponseWriter, r *http.Request) {
+			// no recover here as it will interfere with the handler
+			w.Write([]byte("foobar"))
+			w.(http.Flusher).Flush()
+			// simulate a large response body, some data were written
+			time.Sleep(5 * time.Second)
+			panic(http.ErrAbortHandler)
+		})
+
+		resp, err := client.Get(fmt.Sprintf("https://localhost:%d/writing_and_panicking", port))
+		Expect(err).ToNot(HaveOccurred())
+		body, err := io.ReadAll(resp.Body)
+		Expect(err).To(HaveOccurred())
```

Comment:
What are you asserting here? Does this test now take 5 seconds? That would be way too long.

Reply:
To test the client has encountered stream errors when reading the body. Ideally this should not use sleep. I tried to write a large body, but as soon as I called panic, the response won't even arrive at the client (`client.Get` will return an error).

Reply:
You can block on a channel before panicking, and close that channel once `client.Get` has returned.